1.train_models/log_reg_plates_cp_fs_data/plates_3_3p/nbconverted/plates_3_3p_classify_genotype.py

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Train logistic regressions to classify genotypes for plates 3 and 3 prime
+
+# ## Imports
+
+# In[1]:
+
+
+import sys
+import warnings
+from pathlib import Path
+
+import pandas as pd
+from joblib import dump
+from scipy.stats import uniform
+from sklearn.exceptions import ConvergenceWarning
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import RandomizedSearchCV
+from sklearn.utils import parallel_backend
+
+# ## Find the git root Directory
+
+# In[2]:
+
+
+# Get the current working directory
+cwd = Path.cwd()
+
+if (cwd / ".git").is_dir():
+    root_dir = cwd
+
+else:
+    root_dir = None
+    for parent in cwd.parents:
+        if (parent / ".git").is_dir():
+            root_dir = parent
+            break
+
+# Check if a Git root directory was found
+if root_dir is None:
+    raise FileNotFoundError("No Git root directory found.")
+
+
+# ## Import processing utils
+
+# In[3]:
+
+
+sys.path.append(f"{root_dir}/1.train_models/log_reg_plates_cp_fs_data/utils")
+import log_reg_plates_cp_fs_data_process_split_util as process_split
+
+# ## Define and create paths
+
+# In[4]:
+
+
+# Input paths
+filename3 = "Plate_3_sc_norm_fs.parquet"
+filename3p = "Plate_3_prime_sc_norm_fs.parquet"
+plate_path = Path(
+    f"{root_dir}/nf1_painting_repo/3.processing_features/data/feature_selected_data"
+)
+
+path3 = plate_path / filename3
+path3p = plate_path / filename3p
+
+# Output paths
+models_path = Path("trained_models")
+data_path = Path("model_data")
+
+# Create output paths if nonexistent
+models_path.mkdir(parents=True, exist_ok=True)
+data_path.mkdir(parents=True, exist_ok=True)
+
+
+# ## Generate plate dataframes
+
+# In[5]:
+
+
+# Returns the dataframe returned by the plate 3 parquet file
+plate3df = pd.read_parquet(path3)
+
+# Returns the dataframe returned by the plate 3 prime parquet file
+plate3pdf = pd.read_parquet(path3p)
+
+
+# ## Define column names
+
+# In[6]:
+
+
+well_column = "Metadata_Well"
+gene_column = "Metadata_genotype"
+
+
+# # Preprocess Data
+
+# ## Use only common columns
+
+# In[7]:
+
+
+# Set plate column:
+plate3df["Metadata_plate"] = "3"
+plate3pdf["Metadata_plate"] = "3p"
+
+common_columns = list(plate3df.columns.intersection(plate3pdf.columns))
```

Comment:
```suggestion
common_columns = (plate3df.columns.intersection(plate3pdf.columns).to_list()
```
consider changing this line

Reply:
Changed!